src/main/java/uk/gov/digital/ho/hocs/audit/export/parsers/DataParserFactory.java

```diff
@@ -0,0 +1,43 @@
+package uk.gov.digital.ho.hocs.audit.export.parsers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.stereotype.Service;
+import uk.gov.digital.ho.hocs.audit.application.ZonedDateTimeConverter;
+import uk.gov.digital.ho.hocs.audit.export.caseworkclient.CaseworkClient;
+import uk.gov.digital.ho.hocs.audit.export.infoclient.InfoClientSupplier;
+import uk.gov.digital.ho.hocs.audit.export.parsers.interests.BfInterestDataParser;
+import uk.gov.digital.ho.hocs.audit.export.parsers.interests.FoiInterestDataParser;
+
+@Service
+public class DataParserFactory {
+
+    private final InfoClientSupplier infoClientSupplier;
+    private final CaseworkClient caseworkClient;
+    private final ObjectMapper objectMapper;
+
+    public DataParserFactory(InfoClientSupplier infoClientSupplier,
+                             CaseworkClient caseworkClient,
+                             ObjectMapper objectMapper) {
+        this.infoClientSupplier = infoClientSupplier;
+        this.caseworkClient = caseworkClient;
+        this.objectMapper = objectMapper;
+    }
+
+    public AbstractDataParser getInterestInstance(String caseType, boolean convert, ZonedDateTimeConverter zonedDateTimeConverter)
+        throws IllegalArgumentException
+    {
+        switch (caseType) {
+            case "FOI": {
+                return new FoiInterestDataParser(infoClientSupplier, caseworkClient, objectMapper,
+                        zonedDateTimeConverter, convert);
+            }
+            case "BF": {
+                return new BfInterestDataParser(infoClientSupplier, caseworkClient, objectMapper,
+                        zonedDateTimeConverter, convert);
+            }
+            default:
```

Comment:
As an aside to this I think we could do with some good documentation writing up about creating a new Interest action, as now there are more steps involved in the set up

Reply:
I'm planning on doing a demo and writing some documentation up on it